smallrye-reactive-operators/src/test/java/io/smallrye/reactive/groups/MultiBroadcastTest.java

```diff
@@ -0,0 +1,281 @@
+package io.smallrye.reactive.groups;
+
+import io.smallrye.reactive.Multi;
+import io.smallrye.reactive.helpers.MultiEmitterProcessor;
+import io.smallrye.reactive.operators.MultiAssertSubscriber;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+public class MultiBroadcastTest {
+
+    @Test
+    public void testPublishToAllSubscribers() {
+        MultiEmitterProcessor<Integer> processor = MultiEmitterProcessor.create();
+
+        Multi<Integer> multi = processor.toMulti().broadcast().toAllSubscribers();
+
+        MultiAssertSubscriber<Integer> s1 = multi.subscribe().withSubscriber(MultiAssertSubscriber.create(10));
+        s1.assertHasNotReceivedAnyItem().assertNotTerminated();
+
+        processor.emit(1).emit(2).emit(3);
+        s1.assertReceived(1, 2, 3).assertNotTerminated();
+
+        MultiAssertSubscriber<Integer> s2 = multi.subscribe().withSubscriber(MultiAssertSubscriber.create(1));
+        s2.assertHasNotReceivedAnyItem().assertNotTerminated();
+
+        processor.emit(4).emit(5);
+        // 5 is not received, because we use the lower number of requests.
```

Comment:
I'm not sure what this comment means.

How does `s1` not receive 5, as it appears to be requesting 10 items?

Reply:
no it does not receive 5, because s2 did request only 1 item. Until s2 request the second item, s1 won't see 5. I agree, it's a bit disturbing, but that's the spec...

Reply:
Is that purely because it's in the same process, or if there were two subscribers in different processes one impacts how many the other receives?

Reply:
Yes, the one making the smallest amount of requests wins.
This is because if we retrieve more data, we need to "store" it until all the subscribers consume it. It one consumer does not request anything anymore, and one continues to emit requests and consume data, it would end with an overflow / out of memory.

Reply:
Ah! That makes a lot more sense now.

Is there good education on this, or we will need to educate developers?

I can foresee a system working well with a couple of subscribers and another one is handled with a significantly different item request count and suddenly throughput drops!